src/components/Breadcrumb.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <emqx-breadcrumb separator="/">
+    <transition-group name="breadcrumb" mode="out-in">
+    <emqx-breadcrumb-item v-for="(item,index) in levelList" :key="item.path">
+      <span v-if="item.redirect==='noRedirect'|| index === levelList.length-1" class="no-redirect">{{ $t(`${item.meta.title}`) }}</span>
+      <a v-else @click.prevent="onHandleLink(item)">{{ $t(`${item.meta.title}`) }}</a>
+    </emqx-breadcrumb-item>
+  </transition-group>
+  </emqx-breadcrumb>
+</template>
+
+
+<script lang="ts">
+import { defineComponent, onBeforeMount, ref, watch, reactive, toRefs } from 'vue'
+export default defineComponent({
+  name: 'Breadcrumb'
+})
+</script>
+
+<script lang="ts" setup>
+import { useRoute, useRouter } from 'vue-router'
+
+const $route = useRoute()
+const $router = useRouter()
+const state = reactive({
+  levelList: [] as Array<any>,
+})
+const { levelList } = toRefs(state)
+
+onBeforeMount(() => {
+  getBreadcrumbs()
+})
+
+watch(() => $route.path, () => {
+  getBreadcrumbs()
+}, { immediate: false })
+
+const getBreadcrumbs = () => {
+  const { fullPath, matched } = $route
+  if ($route.meta.hiddenBreadcrumb) {
+    state.levelList = []
+    return
+  }
+  let new_matched = matched.filter(item => item.meta && item.meta.title)
+
+  /**
+   * fixed 返回上级面包屑且路由含有动态参数时找不到路由的问题
+   * 1. 进入页面时，标记 & 保存当前路由的fullpath（包含了路由参数）
+   * 2. 存储上一次匹配的路由，即进入页面前的面包屑
+   * 3. 判断当前路由是否是同父级：
+   *  3.1 同父级下，进去嵌套路由（当前路由 matched 的 length > 上一次路由matched.length）：
+   *      3.11 进入嵌套路由：当前理由标记fullpath，便于下一层返回定位，使用上一次matched + 当前路由
+   *      3.12 返回上一层：在上一次的matched 中定位当前路由位置，截取当前路由之前的数据（包含当前路由）
+   *  3.2 不同父级，直接使用新matched
+   * 对比上一次面包屑列表，如当前路由匹配的列表长度比较大，即进入新页面，则合并 2 步骤的列表 和 当前路由（1已记录fullpath）；否则为返回面包屑上一级，则使用当前页面匹配到的列表
+   */
```

Comment:
Comments are too long and not in Chinese

Reply:
sorry~，forget to remove~